tokenizers/src/pre_tokenizers/metaspace.rs

```diff
@@ -115,29 +118,50 @@ impl Default for Metaspace {
 
 impl PreTokenizer for Metaspace {
     fn pre_tokenize(&self, pretokenized: &mut PreTokenizedString) -> Result<()> {
-        pretokenized.split(|_, mut normalized| {
-            normalized.replace(' ', &self.str_rep)?;
-            match self.prepend_scheme {
-                PrependScheme::Always => {
-                    if !normalized.get().starts_with(self.replacement) {
-                        normalized.prepend(&self.str_rep);
-                    }
+        let result = if self.prepend_scheme == PrependScheme::First {
+            let first = &mut pretokenized.splits[0];
```

Comment:
This will panic if splits is empty, isn't 

```rust
pretokenized.split(|_, mut normalized| {
->
pretokenized.split(|i, mut normalized| {
...
if i == 0
```

Enough for a change ?

Reply:
Also for good measure, the reason it checks offsets.original == 0 instead of `i==0` is because users can send pre pretokenized strings, meanings you can have split n still being "the first token". (Since it's transparent for us at this stage if it was pretiokenized strings from us, or from the users, only the offsets retain that information)